src/main.py

```diff
@@ -0,0 +1,170 @@
+
+        questions = []
+        for question in questionnaire:
+            match question["type"]:
+                case "FREESINGLE":
+                    questions.append(
+                        {
+                            "type": "text",
+                            "name": question["question"],
+                            "message": question["question"],
+                        }
+                    )
+
+                case "FREEMULTIPLE":
+                    # TODO: Deal with 'FREEMULTIPLE' questions.
+                    # For now we ignore if we encounter them.
+                    continue
+
+                case "CHOICESINGLE":
+                    questions.append(
+                        {
+                            "type": "select",
+                            "name": question["question"],
+                            "message": question["question"],
+                            "choices": [opts["answer"] for opts in question["options"]["option"]],
+                        }
+                    )
+                case "CHOICEMULTIPLE":
+                    questions.append(
+                        {
+                            "type": "checkbox",
+                            "name": question["question"],
+                            "message": question["question"],
+                            "choices": [opts["answer"] for opts in question["options"]["option"]],
+                        }
+                    )
+                case _:
+                    raise RuntimeError(f'unexpected question type {question["type"]}')
+
+        questionnaires.append(Questionnaire(name=questionnaire_filepath.stem, questions=questions))
+
+    return questionnaires
+
+
+def load_filled_questionnaires(output_dir: Path) -> Dict[str, Dict[str, str]]:
+    """
+    Loads already (partially) filled in questionnaires from yaml files in the output_dir.
+
+    :param output_dir: Path to a directory containing question and answer yaml files.
+    :return: A dictionary mapping the name of a questionnaire to a dictionary with keys being
+    the question and the value being the answer.
+    :raises: TypeError: If output_dir is not a directory path.
+
+    """
+    if not Path.is_dir(output_dir):
+        raise TypeError(f"{output_dir} must be a directory")
+
+    q_and_a_sources = {}
+    for q_and_a_filepath in output_dir.iterdir():
+        if q_and_a_filepath.suffix != ".yaml":
+            print(f"ignoring unexpected file format {q_and_a_filepath.suffix}", file=sys.stderr)
+            continue
+
+        with open(q_and_a_filepath) as f:
+            q_and_a_yaml = yaml.safe_load(f)
+
+        q_and_a_sources[q_and_a_filepath.stem] = {}
+        for q_and_a in q_and_a_yaml:
+            q_and_a_sources[q_and_a_filepath.stem][q_and_a["question"]] = q_and_a["answer"]
+
+    return q_and_a_sources
+
+
+if __name__ == "__main__":
+    cwd = Path.cwd()
+    question_dir = (cwd / "questionnaires").resolve()
```

Comment:
could you add option handling in here? i always yse argparse. but there are more options

Reply:
Good suggestion, we've added the argparse.